boards/f4vi1/include/periph_conf.h

```diff
@@ -75,24 +82,13 @@ extern "C" {
  * @name UART configuration
  * @{
  */
-#define UART_NUMOF          (1U)
-#define UART_0_EN           1
-#define UART_1_EN           0
-#define UART_IRQ_PRIO       1
+static const uart_conf_t uart_config[] = {
+    /* device, RCC mask, RX pin, TX pin, pin AF, IRQ channel */
+    {USART6, RCC_APB2ENR_USART6EN, GPIO(PORT_C,7), GPIO(PORT_C,6), GPIO_AF8, USART6_IRQn},
+};
 
-/* UART 0 device configuration */
-#define UART_0_DEV          USART6
-#define UART_0_CLKEN()      (RCC->APB2ENR |= RCC_APB2ENR_USART6EN)
-#define UART_0_CLKDIS()     (RCC->APB2ENR &= ~(RCC_APB2ENR_USART6EN))
-#define UART_0_CLK          (84000000)          /* UART clock runs with 84MHz (F_CPU / 2) */
-#define UART_0_IRQ_CHAN     USART6_IRQn
 #define UART_0_ISR          isr_usart6
-/* UART 0 pin configuration */
-#define UART_0_PORT_CLKEN() (RCC->AHB1ENR |= RCC_AHB1ENR_GPIOCEN)
-#define UART_0_PORT         GPIOC
-#define UART_0_TX_PIN       6
-#define UART_0_RX_PIN       7
-#define UART_0_AF           8
+#define UART_NUMOF          (sizeof(uart_config) / sizeof(uart_conf_t))
```

Comment:
any particular reasons for this? I don't mind, just try to understand the benefits...